en/scripting/modules/example.md

```diff
@@ -0,0 +1,252 @@
+
+### Case 4: Using web3
+
+Install the `web3` package in the same way as the `protobufjs` case above. Using the same detection method, determine that `web3` is a **CJS** module. Import in the same way:
+
+```ts
+import web3 from 'web3';
+```
+
+When going back to the editor, a bunch of errors are still present:
+
+![node-module](./example/node-module.png)
+
+This is because the package is customized for Node and contains built-in modules from Node.js, which are not available in Creator, resulting in errors. Usually, these packages also take care of web users. When importing the package directly, the package entry points directly to the Node version, while the web version is placed in the `dist/**.js` or `dist/**.min.js` file under the package. In this case, it is necessary to import the custom version of the Web provided in the package as follows:
+
+```ts
+import web3 from 'web3/dist/web3.min.js';
+```
+
+At this point, the editor does not report errors anymore, but the code prompts: `Could not find a declaration file for module 'web3/dist/web3.min.js'. '/${project_path}/node_modules/web3/dist/web3.min.js' implicitly has an 'any' type.`.
+
+This is because even if these packages have a type specification, it is a type specification for the Node version of that entry. In general, the interface definitions are the same for both versions, just "borrow" the type description from the Node version. Just create an arbitrary new `.d.ts` to the project and write:
+
+```ts
+// Add a type specification for the module "<package name>/dist/**.js"
+declare module "<package name>/dist/**.js" {
+    export * from "<package name>"; // "Steal" the type description of the main entry
+}
+```
+
+> **Note**: currently it is necessary to confirm that if a package is customized for Node. But in the new version of Node, package.json has a nice mechanism added to tell the user which version to use for what case.
+
+### Special case: firebase usage
+
+This is a special case, take a look at the special features of this case. After installing the `firebase` package in the way described above, and analyzing the package according to the `package.json` file, notice the package is in **CJS** format. Based on the entry file, infer that this package is customized for Node (this can be tested according to [Case 4: Using web3](#case-4:-using-web3)), so we have to use the Web custom version `index.esm.js`. The problem is, `index.esm.js` is an **ESM** module, and the Creator recognizes this package as a **CJS** module, but it is also an **ESM** module, which naturally causes errors.
```

Comment:
```suggestion
This is a special case, take a look at the special features of this case. After installing the `firebase` package in the way described above, and analyzing the package according to the `package.json` file, notice the package is in **CJS** format. Based on the entry file, infer that this package is customized for Node (this can be tested according to [Case 4: Using web3](#case-4-using-web3)), so we have to use the Web custom version `index.esm.js`. The problem is, `index.esm.js` is an **ESM** module, and the Creator recognizes this package as a **CJS** module, but it is also an **ESM** module, which naturally causes errors.
```